### PR TITLE
Directory structure and bug clean up

### PR DESCRIPTION
**Pull request by struct**
This is a large diff that introduces a much cleaner directory structure and fixes a few bugs discovered as a result of extensive testing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -238,7 +241,7 @@ ISO_ALLOC_PRINTF_SRC = $(SRC_DIR)/iso_alloc_printf.c
 BUILD_DIR = build
 LDFLAGS = -L$(BUILD_DIR) -lisoalloc
 
-all: library tests
+all: testsm
```
Comment (jvoisin) [inline Makefile:244]: As in "test secure malloc" or "tests malloc"?

File: include/compiler.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/* compiler.h - A secure memory allocator
+ * Copyright 2022 - chris.rohlf@gmail.com */
+
+#pragma once
+
+#define INTERNAL_HIDDEN __attribute__((visibility("hidden")))
+#define ASSUME_ALIGNED __attribute__((assume_aligned(8)))
+#define CONST __attribute__((const))
+
+/* This isn't standard in C as [[nodiscard]] until C23 */
```
Comment (jvoisin) [inline include/compiler.h:10]: We should document the minimum required C version to use/compile isoalloc I think.

File: include/iso_alloc.h [hunk 3]
```diff
@@ -7,6 +7,10 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+#if !__aarch64__ && !__x86_64__
+#pragma message "IsoAlloc is untested and unsupported on 32 bit platforms"
```
Comment (jvoisin) [inline include/iso_alloc.h:11]: Unsupported, or it will simply not work?
Also, maybe we should do something like `assert(sizeof(size_t) >= 64`)` instead?

File: include/iso_alloc_ds.h [hunk 4]
```diff
@@ -0,0 +1,107 @@
+/* iso_alloc_ds.h - A secure memory allocator
+ * Copyright 2022 - chris.rohlf@gmail.com */
+
+#pragma once
```
Comment (jvoisin) [inline include/iso_alloc_ds.h:4]: It this standard and supported everywhere?

File: include/os/android.h [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/* android.h - A secure memory allocator
+ * Copyright 2022 - chris.rohlf@gmail.com */
+
+#pragma once
+
+#include <sys/prctl.h>
+
+#ifndef PR_SET_VMA
+#define PR_SET_VMA 0x53564d41
```
Comment (jvoisin) [inline include/os/android.h:9]: What the hell is this magic number :')
It should be documented.

File: Makefile (hunk 1 above)
Comment (struct) [reply Makefile:244]: As in thats a typo :D

File: include/iso_alloc.h (hunk 3 above)
Comment (struct) [reply include/iso_alloc.h:11]: Yah 32 bit won't work, should be more explicit about that.

File: include/os/android.h (hunk 5 above)
Comment (struct) [reply include/os/android.h:9]: Not my magic number but we can document it https://android.googlesource.com/platform/bionic/+/263325d/libc/include/sys/prctl.h#42

File: src/iso_alloc_util.c [hunk 6]
```diff
@@ -28,7 +28,7 @@ INTERNAL_HIDDEN INLINE int _iso_getcpu(void) {
     uintptr_t a;
     __asm__ volatile("mrs %x0, tpidrro_el0"
                      : "=r"(a)::"memory");
-    return (int) ((a & 0x8) - 1);
+    return (int) ((a & 0x8));
```
Comment (jvoisin) [inline src/iso_alloc_util.c:31]: This change should be part of another commit, since it's not really a cleanup, but a proper change.

File: include/os/android.h (hunk 5 above)
Comment (jvoisin) [reply include/os/android.h:9]: I hate computers.

File: src/iso_alloc_util.c (hunk 6 above)
Comment (struct) [reply src/iso_alloc_util.c:31]: Agree. This slipped in. @devnexen this was the fix in my testing on an M1 macbook but it's not really tested or researched beyond that. Just FYI since you contributed this code.

File: include/iso_alloc_ds.h (hunk 4 above)
Comment (struct) [reply include/iso_alloc_ds.h:4]: Yes https://en.wikipedia.org/wiki/Pragma_once#Portability

File: include/iso_alloc.h [hunk 7]
```diff
@@ -8,7 +8,8 @@
 #include <stdlib.h>
 
 #if !__aarch64__ && !__x86_64__
-#pragma message "IsoAlloc is untested and unsupported on 32 bit platforms"
+#pragma error "IsoAlloc is untested and unsupported on 32 bit platforms"
```
Comment (jvoisin) [inline include/iso_alloc.h:11]: The error will stop the compilation process, making the `assert(sizeof(size_t) >= 64)` moot. You should put it outside of the condition I think.

Comment (struct) [reply include/iso_alloc.h:11]: Also it should be 8, not 64 🤦‍♂️
